src/DotNetty.Handlers/Timeout/ReadTimeoutHandler.cs

```diff
@@ -0,0 +1,227 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace DotNetty.Handlers.Timeout
+{
+    using System;
+    using System.Threading;
+    using DotNetty.Common.Utilities;
+    using DotNetty.Common.Concurrency;
+    using DotNetty.Transport.Channels;
+
+    /// <summary>
+    /// Raises a <see cref="ReadTimeoutException"/> when no data was read within a certain
+    /// period of time.
+    ///
+    /// <pre>
+    /// The connection is closed when there is no inbound traffic
+    /// for 30 seconds.
+    ///
+    /// <example>
+    /// <c>
+    /// var bootstrap = new <see cref="DotNetty.Transport.Bootstrapping.ServerBootstrap"/>();
+    ///
+    /// bootstrap.ChildHandler(new ActionChannelInitializer<ISocketChannel>(channel =>
+    /// {
+    ///     IChannelPipeline pipeline = channel.Pipeline;
+    ///     
+    ///     pipeline.AddLast("readTimeoutHandler", new <see cref="ReadTimeoutHandler"/>(30);
+    ///     pipeline.AddLast("myHandler", new MyHandler());
+    /// } 
+    /// </c>
+    ///            
+    /// <c>
+    /// public class MyHandler : ChannelDuplexHandler 
+    /// {
+    ///     public override void ExceptionCaught(<see cref="IChannelHandlerContext"/> context, <see cref="Exception"/> exception)
+    ///     {
+    ///         if(exception is <see cref="ReadTimeoutException"/>) 
+    ///         {
+    ///             // do somethind
+    ///         }
+    ///         else
+    ///         {
+    ///             base.ExceptionCaught(context, cause);
+    ///         }
+    ///      }
+    /// }
+    /// </c>
+    /// </example>
+    /// 
+    /// <seealso cref="WriteTimeoutHandler"/>
+    /// <seealso cref="IdleStateHandler"/>
+    /// </summary>
+    public class ReadTimeoutHandler : ChannelHandlerAdapter
+    {
+        static readonly long MinTimeoutNanos = TimeSpan.TicksPerMillisecond * TimeUtil.NanosecondPerTick;
+
+        readonly long timeoutNanos;
```

Comment:
yes, I meant to change both name and type. Same story as with Idle timeout handler: Schedule works with TimeSpan so it makes sense to use TimeSpan throughout and therefore avoid conversion on calls to Schedule. Also, judging by idle timeout handler changes, TimeUtil won't be necessary at the end.